presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -3719,4 +3740,11 @@ SchemaTableName getSourceTableName(SchemaTableName table)
         }
         return Optional.empty();
     }
+
+    private static SchemaTableName parseTableName(String tableName)
+    {
+        String[] parts = tableName.split("\\.");
+        checkArgument(parts.length == 2, "Invalid schemaTableName: %s", tableName);
+        return new SchemaTableName(parts[0], parts[1]);
+    }
```

Comment:
Could you add this helper to `SchemaTableName` itself? Check `QualifiedObjectName.valueOf` as an example.

Reply:
Sounds good. Also there are a few other similar helpers in our codebase as follow, should we replace them as well?
- com.facebook.presto.kafka.schema.file.FileTableDescriptionSupplier#parseTableName
```java
    private static SchemaTableName parseTableName(String schemaTableName)
    {
        checkArgument(!isNullOrEmpty(schemaTableName), "schemaTableName is null or is empty");
        List<String> parts = Splitter.on('.').splitToList(schemaTableName);
        checkArgument(parts.size() == 2, "Invalid schemaTableName: %s", schemaTableName);
        return new SchemaTableName(parts.get(0), parts.get(1));
    }
```
- com.facebook.presto.hive.CachingDirectoryLister#parseTableName
- com.facebook.presto.redis.RedisTableDescriptionSupplier#parseTableName

Reply:
Interesting, yes. Actually have two commits for this PR. The first one is to refactor these helpers to use the centralized util. The second commit would be the one you have right now.